server/lib/processDonation.js

```diff
@@ -0,0 +1,134 @@
+const gateways = require('../gateways');
+const constants = require('../constants/transactions');
+const roles = require('../constants/roles');
+const emailLib = require('./email');
+
+module.exports = (Sequelize, donation) => {
+
+  const services = {
+
+    stripe: (donation) => {
+      const group = donation.Group;
+      const user = donation.User;
+      const paymentMethod = donation.PaymentMethod;
+      const subscription = donation.Subscription;
+      var groupStripeAccount, charge;
+
+      return group.getStripeAccount()
```

Comment:
there is a lot of `then` in this method. Can we break it down?

It feels like anything within a `then` should be a one-three liner.